### PR TITLE
Revive test coverage #1140 partly

### PR DESCRIPTION
**Pull request by lzxm160**
work on #1140

### CODE REVIEW

File: action/protocol/context_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2018 IoTeX
```
Comment (zjshen14) [inline action/protocol/context_test.go]: 2019

File: action/protocol/generic_validator_test.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+		require.NoError(err)
+		nselp := action.SealedEnvelope{}
+		require.NoError(nselp.LoadProto(selp.Proto()))
+		err = valid.Validate(c, nselp)
+		require.Error(err)
+		require.True(strings.Contains(err.Error(), "nonce is too low"))
+	}
+}
+
+type MockChainManager struct {
```
Comment (zjshen14) [inline action/protocol/generic_validator_test.go]: Can you use mock_chainmanager.MockChainManager instead?

File: action/protocol/registry_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	p := &MockProtocol{}
+	require.NoError(reg.Register("1", p))
+	// Case I: Normal
+	require.Equal(1, len(reg.All()))
+	// Case II: Registry stores the item which is not a protocol
+	require.NoError(reg.Register("2", nil))
+	require.Panics(func() { reg.All() }, "Registry stores the item which is not a protocol")
+}
+
+type MockProtocol struct {
```
Comment (zjshen14) [inline action/protocol/registry_test.go]: Gen a mock protocol with mock gen pls

Comment (lzxm160) [reply action/protocol/registry_test.go]: caused “import cycle not allowed” error，so I have to build one

**Pull request merged**